### PR TITLE
add match to bare pyteset.raises() - arrays

### PR DESCRIPTION
**Pull request by sathyz**
- [x] ref #30999
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by MomIsBestFriend** [comment]
>     * [x]  closes #30999
> 
>     * [x]  tests added / passed
> 
>     * [x]  passes `black pandas`
> 
>     * [x]  passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
> 
>     * [ ]  whatsnew entry

@sathyz Can you please change the word "closes" to the word "ref"?

File: pandas/tests/arrays/interval/test_ops.py [hunk 1]
```diff
@@ -57,7 +57,7 @@ def test_overlaps_interval_container(self, constructor, other_constructor):
         # TODO: modify this test when implemented
         interval_container = constructor.from_breaks(range(5))
         other_container = other_constructor.from_breaks(range(5))
-        with pytest.raises(NotImplementedError):
+        with pytest.raises(NotImplementedError, match=None):
```
Comment (MomIsBestFriend) [inline pandas/tests/arrays/interval/test_ops.py]:
```suggestion
        with pytest.raises(NotImplementedError, match="^$"):
```

File: pandas/tests/arrays/test_datetimelike.py [hunk 2]
```diff
@@ -774,8 +795,25 @@ def test_casting_nat_setitem_array(array, casting_nats):
...
+        r"Got 'int' instead.|"
+        r"'value' should be a 'Timestamp', 'NaT', or array of those. "
+        r"Got 'int' instead.|"
+        r"'value' should be a 'Period', 'NaT', or array of those. "
+        r"Got 'datetime64' instead.|"
+        r"'value' should be a 'Period', 'NaT', or array of those. "
+        r"Got 'timedelta64' instead.|"
+        r"'value' should be a 'Period', 'NaT', or array of those. "
+        r"Got 'int' instead."
+    )
```
Comment (MomIsBestFriend) [inline pandas/tests/arrays/test_datetimelike.py]: We are trying to use the pattern of:

```python
msg = "|".join(
    [
        "foo",
        "bar,
        "baz"
    ]
)

```

Comment (MomIsBestFriend) [reply pandas/tests/arrays/test_datetimelike.py]: After having a second look, I *think*  we can have this as a single string

```suggestion
    msg = (
        "'value' should be a '(Timestamp|Timedelta|Period)', 'NaT', or array of those. "
        "Got '(timedelta64|datetime64|int)' instead."
    )
```

File: pandas/tests/arrays/test_datetimelike.py [hunk 3]
```diff
@@ -108,14 +108,20 @@ def test_take_fill(self):
         result = arr.take([-1, 1], allow_fill=True, fill_value=pd.NaT)
         assert result[0] is pd.NaT
 
-        with pytest.raises(ValueError):
-            arr.take([0, 1], allow_fill=True, fill_value=2)
+        value = 2
```
Comment (simonjayhawkins) [inline pandas/tests/arrays/test_datetimelike.py]: can you split this out from test_take_fill as a separate test, say test_take_fill_raises and parametrize over value.

File: pandas/tests/arrays/test_datetimelike.py (hunk 2 above)
Comment (sathyz) [reply pandas/tests/arrays/test_datetimelike.py]: assertions fail when I use this,

```
E               AssertionError: Pattern "'value' should be a '(Timestamp|Timedelta|Period)','NaT', or array of those. Got '(timedelta64|datetime64|int)' instead." not found in "'value' should be a 'Timedelta', 'NaT', or array of those. Got 'datetime64' instead."
```



I'm trying to investigate as follows,

```python
>>> import re
>>> msg = "'value' should be a 'Timedelta', 'NaT', or array of those. Got 'datetime64' instead."
>>> pat = r"'value' should be a '(Timestamp|Timedelta|Period)', 'NaT', or array of those. Got '(timedelta64|datetimee64|int)' instead."
>>> re.search(pat, msg)
>>>
```

Comment (simonjayhawkins) [reply pandas/tests/arrays/test_datetimelike.py]: it looks like missing a space after comma

Comment (sathyz) [reply pandas/tests/arrays/test_datetimelike.py]: Ah my bad.

**Comment by jreback** [comment]
thanks @sathyz

**Pull request merged**